dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/Model/RestApiOperation.cs

```diff
@@ -101,6 +104,62 @@ public Uri BuildOperationUrl(IDictionary<string, string> arguments)
         return new Uri($"{serverUrl.TrimEnd('/')}/{path.TrimStart('/')}", UriKind.Absolute);
     }
 
+    /// <summary>
+    /// Renders operation request headers.
+    /// </summary>
+    /// <param name="arguments">The operation arguments.</param>
+    /// <returns>The rendered request headers.</returns>
+    public IDictionary<string, string> RenderHeaders(IDictionary<string, string> arguments)
+    {
+        var headersMetadata = this.Parameters.Where(p => p.Location == RestApiOperationParameterLocation.Header);
+
+        var headers = new Dictionary<string, string>();
+
+        foreach (var header in this.Headers)
+        {
+            var headerName = header.Key;
+            var headerValue = header.Value;
+
+            //A try to resolve header value in arguments.
+            if (arguments.TryGetValue(headerName, out var value))
+            {
+                headers.Add(headerName, value);
+                continue;
+            }
+
+            //Header value is already supplied.
+            if (!string.IsNullOrEmpty(headerValue))
+            {
+                headers.Add(headerName, headerValue);
+                continue;
+            }
+
+            //Getting metadata for the header
+            var headerMetadata = headersMetadata.FirstOrDefault(hm => hm.Name == headerName);
+            if (headerMetadata == null)//No metadata found for the header.
+            {
+                throw new RestApiOperationException($"No value for the '{headerName} header is found.'");
+            }
+
+            //If parameter is required it's value should always be provided.
+            if (headerMetadata.IsRequired)
+            {
+                throw new RestApiOperationException($"No value for the '{headerName} header is found.'");
+            }
+
+            //Parameter is not required and no default value provided.
+            if (string.IsNullOrEmpty(headerMetadata.DefaultValue))
+            {
+                continue;
+            }
+
+            //Using default value.
+            headers.Add(headerName, headerMetadata.DefaultValue);
```

Comment:
How does this work with headers marked as required and also have a default value?

Reply:
According to OpenApi spec - [link](https://swagger.io/docs/specification/describing-parameters/) it does not make sense:
![image](https://user-images.githubusercontent.com/68852919/230026646-9b563d5e-1ba6-4ce2-8790-9f003bb6d786.png)